Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIr.kt

```diff
@@ -0,0 +1,460 @@
+            override val superClassInit: SuperClassInit? = null,
+            override val interfaces: List<StubType> = emptyList(),
+            override val properties: List<PropertyStub> = emptyList(),
+            override val origin: StubOrigin,
+            override val annotations: List<AnnotationStub> = emptyList(),
+            override val childrenClasses: List<ClassStub> = emptyList(),
+            override val companion: Companion? = null,
+            override val functions: List<FunctionalStub> = emptyList(),
+            override val simpleContainers: List<SimpleStubContainer> = emptyList()
+    ) : ClassStub()
+
+    class Companion(
+            override val superClassInit: SuperClassInit? = null,
+            override val interfaces: List<StubType> = emptyList(),
+            override val properties: List<PropertyStub> = emptyList(),
+            override val origin: StubOrigin = StubOrigin.None,
+            override val annotations: List<AnnotationStub> = emptyList(),
+            override val childrenClasses: List<ClassStub> = emptyList(),
+            override val functions: List<FunctionalStub> = emptyList(),
+            override val simpleContainers: List<SimpleStubContainer> = emptyList()
+    ) : ClassStub() {
+        override val companion: Companion? = null
+    }
+
+    class Enum(
+            val classifier: Classifier,
+            val entries: List<EnumEntryStub>,
+            val constructorParams: List<ConstructorParamStub> = emptyList(),
+            override val superClassInit: SuperClassInit? = null,
+            override val interfaces: List<StubType> = emptyList(),
+            override val properties: List<PropertyStub> = emptyList(),
+            override val origin: StubOrigin,
+            override val annotations: List<AnnotationStub> = emptyList(),
+            override val childrenClasses: List<ClassStub> = emptyList(),
+            override val companion: Companion?= null,
+            override val functions: List<FunctionalStub> = emptyList(),
+            override val simpleContainers: List<SimpleStubContainer> = emptyList()
+    ) : ClassStub()
+
+    override val meta: StubContainerMeta = StubContainerMeta()
+
+    override val classes: List<ClassStub>
+        get() = childrenClasses + listOfNotNull(companion)
+
+    override fun accept(visitor: StubIrVisitor) {
+        visitor.visitClass(this)
+    }
+
+    override fun <T, R> accept(visitorAux: StubIrVisitorAux<T, R>, data: T) =
+        visitorAux.visitClass(this, data)
+
+    override val typealiases: List<TypealiasStub> = emptyList()
+}
+
+class ReceiverParameterStub(
+        val type: StubType
+)
+
+class FunctionParameterStub(
+        val name: String,
+        val type: StubType,
+        override val annotations: List<AnnotationStub> = emptyList(),
+        val isVararg: Boolean = false,
+        val origin: StubOrigin = StubOrigin.None
+) : ValueStub(), AnnotationHolder
```

Comment:
A parameter itself is not a value.
Maybe declare `class GetFunctionParameter(val parameter: FunctionParameterStub) : ValueStub()` instead?

Reply:
There is no usages of `FunctionParameterStub` as `ValueStub` left so it can be removed.

Reply:
The same argument is applicable to `ConstructorParamStub`.